cli/src/lockfiles/parsers/pypi.rs

```diff
@@ -94,3 +114,29 @@ fn package(input: &str) -> Option<PackageDescriptor> {
         package_type: PackageType::PyPi,
     })
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn package_with_extras() {
+        assert_eq!(
+            package("celery [ redis ] == 5.0.5"),
```

Comment:
As you mention, `complex-requirements.txt` is a weird test case. I'd go further and say that it needs to be removed in the near future.

So instead, I have added a `Django` example to `requirement.txt`. That file contains only pass test cases, so it seemed like a more logical place to add another pass test case.

I used `Django` instead of your suggestions so that we expand our coverage to as many real world examples as possible.